### PR TITLE
PCL5/netstandard MSBuild support

### PR DESCRIPTION
**Pull request by mhutch**
Adds support for building PCL 5 and .NET Standard MSBuild projects.

See https://gist.github.com/mhutch/44e0501bfafa4fd22b4c00dad2684260 for an example.

### CODE REVIEW

File: mcs/tools/xbuild/data/4.0/Microsoft.Common.targets [hunk 1]
```diff
@@ -49,7 +49,9 @@
 		<_OriginalPlatform>$(Platform)</_OriginalPlatform>
 		<Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
 		<PlatformName Condition="'$(PlatformName)' == ''">$(Platform)</PlatformName>
+	</PropertyGroup>
 
+	<!-- in MSBuild, these properties are set in a separate file that is only imported for .NETFramework -->
```
Comment (Therzok) [inline mcs/tools/xbuild/data/4.0/Microsoft.Common.targets]: Missing the propertygroup line

Comment (mhutch) [inline mcs/tools/xbuild/data/4.0/Microsoft.Common.targets]: Fixed!

File: mcs/class/Microsoft.NuGet.Build.Tasks/Makefile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+CLEAN_FILES = Microsoft.NuGet.Build.Tasks.Strings.resources
+
+EXTRA_DISTFILES = \
+	$(NUGET_BUILDTASKS_REPO_DIR)/src/Microsoft.NuGet.Build.Tasks/Strings.resx \
+	$(NUGET_BUILDTASKS_REPO_DIR)/build/PublicKey.snk
+
+include ../../build/library.make
+
+$(build_lib): Microsoft.NuGet.Build.Tasks.Strings.resources
```
Comment (marek-safar) [inline mcs/class/Microsoft.NuGet.Build.Tasks/Makefile]: Use `RESOURCE_DEFS` instead

File: mcs/tools/xbuild/Makefile [hunk 3]
```diff
@@ -20,18 +20,29 @@ else
...
+PCL5_FX_SRC=../../../external/buildtools/src/Portable/Frameworks/v5.0
+
 NETFRAMEWORK_DIR=$(mono_libdir)/mono/xbuild-frameworks/.NETFramework
+PCL5_FX_DIR=$(mono_libdir)/mono/xbuild-frameworks/.NETPortable/v5.0
 VS_TARGETS_DIR = $(mono_libdir)/mono/xbuild/Microsoft/VisualStudio
 PORTABLE_TARGETS_DIR = $(mono_libdir)/mono/xbuild/Microsoft/Portable
+NUGET_BUILDTASKS_TARGETS_DIR = $(mono_libdir)/mono/xbuild/Microsoft/NuGet
 
-
+ifeq (14.0, $(XBUILD_VERSION))
```
Comment (marek-safar) [inline mcs/tools/xbuild/Makefile]: What about removing ifeq from here and make the targets no-op for non v14 versions

File: mcs/class/Microsoft.NuGet.Build.Tasks/Makefile (hunk 2 above)
Comment (mhutch) [inline mcs/class/Microsoft.NuGet.Build.Tasks/Makefile]: Fixed

File: mcs/tools/xbuild/Makefile (hunk 3 above)
Comment (mhutch) [inline mcs/tools/xbuild/Makefile]: I reorganized the targets to make it clearer which are versioned and which are global. The global ones are only installed for latest (v14).

File: mcs/class/Microsoft.NuGet.Build.Tasks/Makefile (hunk 2 above)
Comment (mhutch) [inline mcs/class/Microsoft.NuGet.Build.Tasks/Makefile]: Looks like `RESOURCE_DEFS` is broken on xbuild profiles, I'll take a look at fixing it tomorrow

**Pull request merged**